core/base/v7/inc/ROOT/TDirectory.h

```diff
@@ -96,47 +92,51 @@ class TDirectory {
   /// \param name  - Key of the object.
   /// \param args  - arguments to be passed to the constructor of `T`
   template <class T, class... ARGS>
-  std::weak_ptr<T> Create(const std::string& name, ARGS... args) {
-    auto ptr = std::make_shared<ToContentType_t<T>>(std::forward<ARGS...>(args...));
+  decltype(auto) Create(const std::string& name, ARGS&&... args) {
+    auto ptr = std::make_shared<ToContentType_t<T>>(std::forward<ARGS>(args)...);
     Add(name, ptr);
     return ptr;
   }
 
-  /// Find the TDirectoryEntryPtrBase associated to the name. Returns nullptr if
+  /// Find the TDirectoryEntry associated to the name. Returns empty TDirectoryEntry if
   /// nothing is found.
-  const Internal::TDirectoryEntryPtrBase* Find(const std::string& name) const {
+  // FIXME: CXX17: return std::optional<TDirectoryEntry>
```

Comment:
Now I'm not so sure we need `std::optional`. `TDirectoryEntry` already has sematics of being empty. So I dont see reasons using `optional` except for "because I can".

Yet marker stule still might be useful somewhere.